src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Resolve.java

```diff
@@ -1052,8 +1052,11 @@ public MethodCheckContext(boolean strict, DeferredAttrContext deferredAttrContex
 
         public boolean compatible(Type found, Type req, Warner warn) {
             InferenceContext inferenceContext = deferredAttrContext.inferenceContext;
+            boolean anyIsUndetVar = inferenceContext.asUndetVar(found).hasTag(UNDETVAR) || inferenceContext.asUndetVar(req).hasTag(UNDETVAR);
+            boolean anyIsPrimitiveClass = found.isPrimitiveClass() || req.isPrimitiveClass();
             return strict ?
-                    types.isSubtypeUnchecked(inferenceContext.asUndetVar(found), inferenceContext.asUndetVar(req), warn) :
+                    (anyIsUndetVar && anyIsPrimitiveClass ? false :
```

Comment:
Could this code be simplified if we hacked directly into the subtyping routine? Note that, I believe that would be necessary anyway, because you have to take into account cases like:

`List<Point> <: List<T?>`

and, also:

`List<Point> <: List<? extends T?>`

So, I think fixing at the level of subtyping/type-containment is what we want here.

Reply:
sure, I will try this